packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -354,49 +360,82 @@ export class Pipeline extends PipelineBase {
     ];
   }
 
-  private ensureReplicationBucketExistsFor(region?: string): s3.IBucket {
-    if (!region) {
-      return this.artifactBucket;
+  private ensureReplicationBucketExistsFor(action: IAction): CrossRegionInfo {
+    const pipelineStack = Stack.of(this);
+
+    let actionRegion: string | undefined;
+    let otherStack: Stack | undefined;
+
+    const actionResource = action.actionProperties.resource;
+    if (actionResource) {
+      const actionResourceStack = Stack.of(actionResource);
```

Comment:
should we validate that `actionProperties.region` is the same or undefined?

Reply:
Like I said before, I don't think that's a great idea. This could be passed by a concrete action class, which the customer has no control over, and I wouldn't be able to get rid of the validation error.

Reply:
ok